pepkit/looper
# Summarize features

This PR implements a major new looper feature, rehauling `looper summarize`. there are 2 components:

- implements custom summarizers
- implements project summary HTML output

Custom summarizers can now be specified in the `pipeline_interface.yaml` file. `looper summarize` will run these summarizers in addition to making the script. Check out `pepatac/dev` for an example of how this works.

In addition, the `pipeline_interface.yaml` also includes information about what the summarizer output is. This information is used, along with any figures (now called objects) produced by the pipeline, to make a slick HTML project report, which includes all the project-level figures and sample-level figures.


To test this:
-install `looper` off this branch.
-make sure you're using the latest version of pypiper (>=0.8), which is now on pypi under `piper`.
-test a project with a pipeline that uses the new `report_object` function, or add this to your pipeline (see:
http://pypiper.readthedocs.io/en/latest/develop-statistics.html#reporting-objects and 
http://pypiper.readthedocs.io/en/latest/api.html#pypiper.manager.PipelineManager.report_object)
-now, try adding a `summarizer` (see #30) for the pipeline. looper summarize will run this. You can see the docs for how this works in this PR: https://github.com/pepkit/looper/pull/34/files#diff-8755f75929e8d131d4de5340a69bcc0c
